builtin/game/falling.lua

```diff
@@ -146,13 +146,28 @@ end
 
 local function drop_attached_node(p)
 	local n = core.get_node(p)
+	local node_meta, meta_fields = core.get_meta(p), nil
+	if node_meta then meta_fields = node_meta:to_table().fields end
 	core.remove_node(p)
 	for _, item in pairs(core.get_node_drops(n, "")) do
 		local pos = {
 			x = p.x + math.random()/2 - 0.25,
 			y = p.y + math.random()/2 - 0.25,
 			z = p.z + math.random()/2 - 0.25,
 		}
+		-- If it drops itself, preserve its metadata.
+		if item == n and meta_fields then
+			local stack, stack_meta = ItemStack({name=n, count=1}), nil
+			if stack then stack_meta = stack:get_meta() end
```

Comment:
why would stack be `nil` here?

Reply:
Defensive programming.  Is it possible for n to be invalid in some way?  Perhaps ""?

Reply:
I think it's reasonable to assume that `core.get_node_drops()` does not return a nil drop

(also no `end` here either)